src/components/NewDatePicker/index.native.js

```diff
@@ -0,0 +1,149 @@
+        this.setState({selectedDate});
+        this.props.onInputChange(moment(selectedDate).format(CONST.DATE.MOMENT_FORMAT_STRING));
+
+        this.hidePicker();
+    }
+
+    render() {
+        const dateAsText = this.props.value || this.props.defaultValue ? moment(this.props.value || this.props.defaultValue).format(CONST.DATE.MOMENT_FORMAT_STRING) : '';
+        return (
+            <View ref={ref => this.wrapperRef = ref} style={[styles.flex2]}>
+                <TextInput
+                    forceActiveLabel
+                    label={this.props.label}
+                    value={dateAsText}
+                    placeholder={this.props.placeholder}
+                    errorText={this.props.errorText}
+                    containerStyles={this.props.containerStyles}
+                    textInputContainerStyles={this.state.isPickerVisible ? [styles.borderColorFocus] : []}
+                    onPress={this.showPicker}
+                    editable={false}
+                    disabled={this.props.disabled}
+                    onBlur={this.props.onBlur}
+                    ref={(el) => {
+                        if (!_.isFunction(this.props.innerRef)) {
+                            return;
+                        }
+                        if (el && el.focus && typeof el.focus === 'function') {
+                            let inputRef = {...el};
+                            inputRef = {...inputRef, focus: this.showPicker};
+                            this.props.innerRef(inputRef);
+                            return;
+                        }
+
+                        this.props.innerRef(el);
+                    }}
+                />
+                {this.state.isPickerVisible && (
+                <View style={[styles.datePickerPopover, styles.border]}>
+                    <CalendarPicker
+                        minDate={this.minDate}
+                        maxDate={this.maxDate}
+                        value={this.state.selectedDate}
+                        onSelected={this.updateLocalDate}
+                        onChanged={this.props.onDateChanged}
+                        defaultMonth={this.props.defaultMonth}
+                        defaultYear={this.props.defaultYear}
+                        onClosePressed={this.hidePicker}
+                    />
+                </View>
+                )}
+            </View>
+        );
+    }
+}
+
+NewDatePicker.propTypes = datepickerPropTypes;
+NewDatePicker.defaultProps = defaultProps;
+
+/**
+ * We're applying localization here because we present a modal (with buttons) ourselves
+ * Furthermore we're passing the locale down so that the modal and the date spinner are in the same
+ * locale. Otherwise the spinner would be present in the system locale and it would be weird if it happens
+ * that the modal buttons are in one locale (app) while the (spinner) month names are another (system)
+ */
+export default compose(
+    withLocalize,
```

Comment:
so the web version does not require the localization?

Reply:
localization is not needed at all